addons/sourcemod/scripting/nd_building_fixes.sp

```diff
@@ -4,138 +4,213 @@
 #include <nd_rounds>
 #include <nd_struct_eng>
 #include <nd_stocks>
+#include <nd_structures>
+#include <nd_commander_build>
 
 //Version is auto-filled by the travis builder
-public Plugin myinfo = 
+public Plugin myinfo =
 {
-	name 		= "[ND] Buildings Fixes",
-	author 		= "stickz",
-    description	= "Prevent building walls inside relays/repeaters",
-    version 	= "dummy",
-	url 		= "https://github.com/stickz/Redstone/"
+    name        = "[ND] Buildings Fixes",
+    author      = "stickz, databomb",
+    description = "Prevent building walls inside relays/repeaters",
+    version     = "dummy",
+    url         = "https://github.com/stickz/Redstone/"
 }
 
+char g_sPowerStructErrorMessage[2][64] = {"CANNOT BUILD TOO NEAR WIRELESS REPEATER.", "CANNOT BUILD TOO NEAR RELAY TOWER."};
+
 /* Auto-Updater Support */
 #define UPDATE_URL  "https://github.com/stickz/Redstone/raw/build/updater/nd_building_fixes/nd_building_fixes.txt"
 #include "updater/standard.sp"
 
 #define DISTANCE_INSIDE_RELAY 60
 
-public void OnPluginStart() {
-	AddUpdaterLibrary(); //auto-updater	
+// Check for nd_structure_intercept being available
+bool g_bStructureDetourAvailable = true;
+public void OnAllPluginsLoaded()
+{
+    g_bStructureDetourAvailable = LibraryExists("nd_structure_intercept");
+    if (!g_bStructureDetourAvailable)
+    {
+        LogError("Problem was found and automatically reverting to ND_OnStructureCreated. Check gamedata.");
```

Comment:
Needs clarity. How about: `Structure detour not available. Reverting to ND_OnStructureCreated. Check gamedata.`